### PR TITLE
Add configuration for new notifications check

### LINKED ISSUE #5881: Add option to disable/setInterval of the notification polling
Every 30 seconds there is a request to /admin/notification/find-last-unread

This can make debugging extremely frustrating.

Please add an option to disable this or at least let me set the interval to a really high number.

### PR DESCRIPTION
**Pull request by storbahn**
Add configuration to activate and set interval for new notifications check

  
  

## Changes in this pull request  
Resolves #

### CODE REVIEW

**Comment by jdreesen** [comment]
I think this fixes #5881

File: bundles/CoreBundle/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -2276,4 +2277,36 @@ private function addGlossaryNode(ArrayNodeDefinition $rootNode): void
...
+    {
+        $rootNode
+            ->children()
+                ->arrayNode('notifications')
+                    ->addDefaultsIfNotSet()
+                    ->children()
+                        ->arrayNode('check_new_notification')
+                            ->addDefaultsIfNotSet()
+                            ->children()
+                                ->booleanNode('active')
```
Comment (brusch) [inline bundles/CoreBundle/DependencyInjection/Configuration.php]: I'd prefer `canBeDisabled()` here, to have it consistent with other options.

File: bundles/CoreBundle/DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -2276,4 +2277,36 @@ private function addGlossaryNode(ArrayNodeDefinition $rootNode): void
                             ->useAttributeAsKey('name')
                             ->prototype('scalar');
     }
+    
+    /**
+     * @param ArrayNodeDefinition $rootNode
+     */
+    private function addNotificationsNode(ArrayNodeDefinition $rootNode): void
```
Comment (brusch) [inline bundles/CoreBundle/DependencyInjection/Configuration.php]: I think we should probably put this into the `AdminBundle` where it belongs to.

**Review by brusch — CHANGES_REQUESTED** [review]

File: bundles/CoreBundle/DependencyInjection/Configuration.php (hunk 1 above)
Comment (storbahn) [reply bundles/CoreBundle/DependencyInjection/Configuration.php]: updated.

File: bundles/CoreBundle/DependencyInjection/Configuration.php (hunk 2 above)
Comment (storbahn) [reply bundles/CoreBundle/DependencyInjection/Configuration.php]: updated.

File: bundles/AdminBundle/DependencyInjection/Configuration.php [hunk 3]
```diff
@@ -267,4 +254,26 @@ protected function buildDocumentsNode()
...
+        $notificationsNode = $treeBuilder->getRootNode();
+
+        $notificationsNode
+            ->addDefaultsIfNotSet()
+            ->children()
+                ->arrayNode('check_new_notification')
+                    ->canBeDisabled()
+                    ->children()
+                        ->integerNode('interval')
+                            ->defaultValue(30000)
```
Comment (jdreesen) [inline bundles/AdminBundle/DependencyInjection/Configuration.php]: How about changing this to seconds? I don't think millisecond precision is needed here, and it's easier for the user in seconds.

Comment (storbahn) [reply bundles/AdminBundle/DependencyInjection/Configuration.php]: updated.

**Comment by brusch** [comment]
Changed base to `11.x` ...

**Comment by mcop1** [comment]
@storbahn thanks for the pr, looks good. 
Maybe we can add 2 more things:

- We could add the enabled check here too https://github.com/pimcore/pimcore/blob/7702e75426a53a63c21f6fb2049c48841e2a028c/bundles/AdminBundle/Resources/public/js/pimcore/layout/toolbar.js#L1213 . With that, the menu wouldn´t be added to the toolbar, if the notifications aren´t enabled.
- It would be really great and appreciated, if you could add 2 or 3 sentences to the documentation on how you can enable/disable this feature

Thank you for your efforts!

**Comment by mcop1** [comment]
@storbahn any chance you had the time to look into my comment?

**Comment by storbahn** [comment]
@mcop1 I updated this PR.

I hope I understand you correctly. Could you please check it again?

File: bundles/AdminBundle/src/Controller/Admin/IndexController.php [hunk 4]
```diff
@@ -405,6 +406,25 @@ protected function addCustomViewSettings(array &$settings)
         return $this;
     }
 
+    /**
+     * @param array $settings
+     * @param Config $config
+     *
+     * @return $this
+     */
+    protected function addNotificationSettings(array &$settings, Config $config)
```
Comment (blankse) [inline bundles/AdminBundle/src/Controller/Admin/IndexController.php]:
```suggestion
    /**
     * @return $this
     */
    protected function addNotificationSettings(array &$settings, Config $config): static
```

File: doc/Development_Documentation/18_Tools_and_Features/41_Notifications.md [hunk 5]
```diff
@@ -110,3 +110,43 @@ public function defaultAction(
 }
 ```
 
+### Configuration
+
+The configuration allow to enable/disable the *Notifications menu* in the toolbar
+and the *Check of new notifications* (url: /admin/notification/find-last-unread).
+It takes place in the `pimcore_admin` configuration within the section `notifications`.
```
Comment (jdreesen) [inline doc/Development_Documentation/18_Tools_and_Features/41_Notifications.md]:
```suggestion
The configuration allows enabling/disabling the *Notifications menu* in the toolbar
and the *check of new notifications* (url: `/admin/notification/find-last-unread`).
It is done in the `pimcore_admin` configuration in the `notifications` section.
```

File: doc/Development_Documentation/18_Tools_and_Features/41_Notifications.md [hunk 6]
```diff
@@ -110,3 +110,43 @@ public function defaultAction(
...
+
+* Disable only the check of new unread notifications:
+```yaml
+pimcore_admin:
+    notifications:
+        check_new_notification:
+            enabled: false
+```
+
+* Change interval to get new unread notifications:
```
Comment (jdreesen) [inline doc/Development_Documentation/18_Tools_and_Features/41_Notifications.md]:
```suggestion
* Change interval to fetch new unread notifications (in seconds):
```

**Comment by mcop1** [comment]
@storbahn Yes, thank you, works really well. Could you please have one last look at the merge conflict (and sorry for the late response)?

**Comment by storbahn** [comment]
@mcop1 merge conflicts resolved

**Pull request merged**